components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/DataSourceManager.java

```diff
@@ -127,48 +126,42 @@ public void initDataSources(String configurationDir, Map<String, DataSourceReade
         if (dataSourceReaders.isEmpty()) {
             throw new RuntimeException("No data source readers found. Data sources will not be initialized!");
         }
+
+        DataSourcesConfiguration dataSourceConfiguration = null;
         try {
-            Path dataSourcesPath = Paths.get(configurationDir);
-            File dataSourcesFolder = dataSourcesPath.toFile();
-            File[] dataSourceConfigFiles = dataSourcesFolder.listFiles();
-
-            if (dataSourceConfigFiles != null) {
-                for (File dataSourceConfigFile : dataSourceConfigFiles) {
-                    if (dataSourceConfigFile.getName().endsWith(FILE_NAME_SUFFIX)) {
-                        initDataSource(dataSourceConfigFile);
-                    }
+            // check whether datasource is configured in deployment.yaml. create datasource only if configuration
+            // exists in deployment.yaml
+            if (configProvider.getConfigurationObject(WSO2_DATASOURCES_NAMESPACE) != null) {
+                dataSourceConfiguration = configProvider.getConfigurationObject(DataSourcesConfiguration.class);
+                if (dataSourceConfiguration.getDataSources() == null && dataSourceConfiguration.getDataSources()
+                    .isEmpty()) {
+                    throw new DataSourceException("configuration doesn't specify any datasource configurations");
```

Comment:
Would it be more meaningful to make the first letter of the word "configuration" uppercase?